### DISCUSSION TITLE
Multiple help?

### DISCUSSION THREAD

**Discussion by aviramha** [zkat/miette#227] [Q&A] [answered]
Hi!
I'm trying to use miette and I have a use case where an Enum implements `Diagnostic` (using the derive) and we provide it with `help` attributes to the variants. We also want a "general" help that is always displayed when there's an error.
I tried wrapping the error with another struct that has `help` attribute of it's own, but this doesn't seem to work.
Is it possible to achieve my plan or any recommendation on how to do so?
I thought about implementing a ReportHandler but this seems like an overkill..

**Comment by zkat** [comment, marked as answer]
If you're willing to do a wrapper, you could use something like this, if I understand what you're asking for:

```rust
struct WrapperErr(MyErrEnum);

impl trait miette::Diagnostic for WrapperErr {
  fn help<'a>(&'a self) -> Option<Box<dyn Display + 'a>> {
    let extra_help = "For more help please check out our documentation at https://github.com/example/example";
    if let Some(help) = self.0.help() {
      Some(Box::new(format!("{help}\n\n{extra_help}")))
    } else {
      Some(Box::new(extra_help))
    }
  }
  fn code<'a>(&'a self) -> Option<Box<dyn Display + 'a>> {
    self.0.code()
  }
  fn severity(&self) -> Option<miette::Severity> {
    self.0.severity()
  }
  // ... etc, for all other protocol methods.
}
```
